### DISCUSSION TITLE
why execute script in codeql workflow like that way

### DISCUSSION THREAD

**Discussion by taojintianxia** [apache/skywalking#10977] [Q&A] [answered]
hi dear community, I was reading the codeql workflow and I found following script makes me curious

```
- run: ./mvnw -q -Dmaven.test.skip=true clean install || ./mvnw -q -Dmaven.test.skip=true clean install
```

what does this script mean ? why not just run one command ?

**Comment by Superskyyy** [comment]
@taojintianxia Not sure but I guess it's meant for a redundancy for flaky builds. IDK if its still relevant.

**Reply by kezhenxu94** [reply, marked as answer]
> @taojintianxia Not sure but I guess it's meant for a redundancy for flaky builds. IDK if its still relevant.

Bingo, `xxx || xxx` retries the command `xxx` it fails. I guess it can be removed as the build looks to be stable for a long time, but leaving it as is does no harm tho.
